topo/node/lemming/lemming.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package lemming contains a node implementation for a lemming device.
+package lemming
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openconfig/kne/topo/node"
+
+	tpb "github.com/openconfig/kne/proto/topo"
+)
+
+func New(nodeImpl *node.Impl) (node.Node, error) {
+	if nodeImpl == nil {
+		return nil, fmt.Errorf("nodeImpl cannot be nil")
+	}
+	if nodeImpl.Proto == nil {
+		return nil, fmt.Errorf("nodeImpl.Proto cannot be nil")
+	}
+	defaults(nodeImpl.Proto)
+	n := &Node{
+		Impl: nodeImpl,
+	}
+	return n, nil
+}
+
+type Node struct {
+	*node.Impl
+}
+
+func (n *Node) ResetCfg(ctx context.Context) error {
```

Comment:
PushConfig()
ResetCfg()
GenerateSelfSigned()

should all be implemented even if its just to return an "unimplemented" error with a message describing how to do the task (even if its as simple as config reset is not supported by Lemming, or config reset can be done by opening a cli on Lemming container and doing ...)

and you should include a block like: https://github.com/openconfig/kne/blob/f268e9f5c7adf28e4529dfa5aefa911124d9da23/topo/node/ceos/ceos.go#L51 to validate the signatures